### PR TITLE
chore(sdk): add NetworkPrimitives for NetworkManager

### LINKED ISSUE #12517: Introduce `NetworkPrimitives` generic to `NetworkManager`
### Describe the feature

PRs https://github.com/paradigmxyz/reth/pull/12435, https://github.com/paradigmxyz/reth/pull/12469, https://github.com/paradigmxyz/reth/pull/12481 introduced `NetworkPrimitives` generic and started integrating network components with it

We need to integrate it into `NetworkManager` and `NetworkHandle` now:
https://github.com/paradigmxyz/reth/blob/8677514bc2ed70499b315b14104ac1c9e39b549b/crates/net/network/src/manager.rs#L79
https://github.com/paradigmxyz/reth/blob/8677514bc2ed70499b315b14104ac1c9e39b549b/crates/net/network/src/network.rs#L42

Recommended steps are:
1. Add `N: NetworkPrimitives` to `NetworkManager` and all its implementations.
2. Pass generic to its members which are already generic over NetworkPrimitives. Such members are `Swarm`, `BlockImport`, `NetworkEvent`.
3. Add generic to other types which NetworkManager relies on and which require NetworkPrimitives to compile

**Note**:  We want to avoid introducing generics to types holding NetworkManager/NetworkHandle for now to keep scope smaller. Those types should be kept as is and should rely on default EthNetworkPrimitives generic value on NetworkManager

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by lakshya-sky**
closes #12517

### CODE REVIEW

**Comment by klkvr** [comment]
you should also add generic to its implementations, eg here:

https://github.com/paradigmxyz/reth/blob/68a6ada460c31c74f25d3b5418b8bc878edfe082/crates/net/network/src/manager.rs#L119

this would highlight some more places where generic is required

**Comment by lakshya-sky** [comment]
@klkvr could you help me a bit here? I've updated all the places but it seems I can't keep up with the chain of all generics.

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/net/network-api/src/events.rs
Comment (mattsse) [inline crates/net/network-api/src/events.rs:1]: this I'd like to change separately

**Comment by klkvr** [comment]
@lakshya-sky the issue is that here stream expects a stream of `NodeEvent`s with default generic but it is actually generic over `NetworkEventListenerProvider::NetworkPrimitives`

I think for now it should be OK to remove NetworkPrimitives generic from NodeEvent and instead just convert `Established` events into `Other` variant

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**